src/ray/raylet/local_object_manager.cc

```diff
@@ -271,44 +281,83 @@ void LocalObjectManager::AddSpilledUrls(
-    // be retrieved by other raylets.
-    RAY_CHECK_OK(object_info_accessor_.AsyncAddSpilledUrl(
-        object_id, object_url, node_id_object_spilled, it->second->GetSize(),
-        [this, object_id, object_url, callback, num_remaining](Status status) {
-          RAY_CHECK_OK(status);
-          // Unpin the object.
-          auto it = objects_pending_spill_.find(object_id);
-          RAY_CHECK(it != objects_pending_spill_.end());
-          num_bytes_pending_spill_ -= it->second->GetSize();
-          objects_pending_spill_.erase(it);
-
-          // Update the object_id -> url_ref_count to use it for deletion later.
-          // We need to track the references here because a single file can contain
-          // multiple objects, and we shouldn't delete the file until
-          // all the objects are gone out of scope.
-          // object_url is equivalent to url_with_offset.
-          auto parsed_url = ParseURL(object_url);
-          const auto base_url_it = parsed_url->find("url");
-          RAY_CHECK(base_url_it != parsed_url->end());
-          if (!url_ref_count_.contains(base_url_it->second)) {
-            url_ref_count_[base_url_it->second] = 1;
-          } else {
-            url_ref_count_[base_url_it->second] += 1;
-          }
-          spilled_objects_url_.emplace(object_id, object_url);
-
-          (*num_remaining)--;
-          if (*num_remaining == 0 && callback) {
-            callback(status);
-          }
-        }));
+    if (RayConfig::instance().ownership_based_object_directory_enabled()) {
+      // TODO(Clark): Don't send RPC to owner if we're fulfilling an owner-initiated
+      // spill RPC.
+      rpc::AddSpilledUrlRequest request;
+      request.set_object_id(object_id.Binary());
+      request.set_spilled_url(object_url);
+      request.set_spilled_node_id(node_id_object_spilled.Binary());
+      request.set_size(it->second.first->GetSize());
+
+      auto owner_client = owner_client_pool_.GetOrConnect(it->second.second);
+      RAY_LOG(DEBUG) << "Sending spilled URL " << object_url << " for object "
+                     << object_id << " to owner "
+                     << WorkerID::FromBinary(it->second.second.worker_id());
+      // Send spilled URL, spilled node ID, and object size to owner.
+      owner_client->AddSpilledUrl(
+          request, [unpin_callback, object_id](Status status,
+                                               const rpc::AddSpilledUrlReply &reply) {
+            unpin_callback(status);
+          });
+    } else {
+      // Write to object directory. Wait for the write to finish before
+      // releasing the object to make sure that the spilled object can
+      // be retrieved by other raylets.
+      RAY_CHECK_OK(object_info_accessor_.AsyncAddSpilledUrl(
+          object_id, object_url, node_id_object_spilled, it->second.first->GetSize(),
+          unpin_callback));
+    }
   }
-}
+}  // namespace raylet
```

Comment:
What's this for?

Reply:
I think my editor bugged out and added this, I'll remove it.